taxonomy/utils.py

```diff
@@ -176,9 +221,20 @@ def refresh_product_skills(products, should_commit_to_db, product_type):
         else:
             skill_attr_val = product[skill_extraction_attr]
         if skill_attr_val:
-            translated_skill_attr = get_translated_skill_attribute_val(
-                product[key_or_uuid], skill_attr_val, product_type
-            )
+            # check if request should be processed as well as pass additional data
+            extra_data, skip = process_skill_attr(skill_attr_val, product[key_or_uuid], product_type)
```

Comment:
Instead of having one function to do two jobs:

1. Get the data
2.  Figure out if the processing should be skipped.

It will make more sense to have two functions, one which calculate extra_data or new_data and the other function probablly uses that to calculate if we need to skip processing.

Reply:
Makes sense. Updated